src/notebooks/phdi_analytics.py

```diff
@@ -0,0 +1,297 @@
+
+
+def geocode(row):
+    """
+    Placeholder geocoder for now, real data would come geocoded
+
+    Comment for the future -- it would be helpful, in addition,
+    to geocoding with SmartyStreets, to also pull in data,
+    e.g. socioeconomic data, that can be joined at census geography
+    levels (e.g. https://www.neighborhoodatlas.medicine.wisc.edu/
+    Area Deprivation Index created at Census Block Group Level)
+    """
+    index, lat, lng = row
+    try:
+        census = cg.coordinates(lng, lat)['2020 Census Blocks'][0]
+
+        data = dict(geoid=census['GEOID'],
+                    state=census['STATE'],
+                    county=census['COUNTY'],
+                    tract=census['TRACT'],
+                    block=census['BLOCK'],
+                    lat=lat,
+                    lng=lng)
+
+    except Exception:
+        data = dict(lat=lat, lng=lng)
+
+    return data
+
+
+def generate_point(center: Point, radius: int) -> Point:
+    """
+    This is from
+
+    https://stackoverflow.com/questions/31192451/
+    """
+    radius_in_kilometers = radius * 1e-3
+    random_distance = random.random() * radius_in_kilometers
+    random_bearing = random.random() * 360
+    return geodesic(kilometers=random_distance).destination(center, random_bearing)
+
+
+def write_points(points, filename='synthetic_locations.csv'):
+    """
+    Output a CSV so that you can upload to Google Maps
+
+    This is to just check the synthetic points
+    """
+    output_file = os.path.join(OUTPUTS_DIR, filename)
+    with open(output_file, 'w', encoding='UTF8', newline="") as f:
+        writer = csv.writer(f)
+        writer.writerow(['latitude', 'longitude'])
+        for p in points:
+            writer.writerow([p[0], p[1]])
+
+
+def generate_random_date(start_date, end_date):
+    time_between_dates = end_date - start_date
+    days_between_dates = time_between_dates.days
+    random_number_of_days = random.randrange(days_between_dates)
+    random_date = start_date + dt.timedelta(days=random_number_of_days)
+
+    return random_date
+
+
+def generate_covid_positive(row):
+    if row['is_fully_vax'] == 0:
+        covid_positive = random.choices([0, 1], weights=(0.97, 0.03), k=1)
+        return covid_positive[0]
+    else:
+        covid_positive = random.choices([0, 1], weights=(0.99, 0.01), k=1)
+        return covid_positive[0]
+
+
+def create_dataframe(radius, number_of_points, center):
+    start_date = dt.date(2021, 6, 1)
+    end_date = dt.date(2022, 3, 31)
+
+    points = [generate_point(center, radius) for _ in range(number_of_points)]
+
+    # Generating the synthetic data as a dataframe
+    latitudes = []
+    longitudes = []
+    personids = random.sample(range(len(points)), len(points))
+    first_names = []
+    last_names = []
+    addresses = []
+    fake = Faker()
+    zip_codes = []
+    event_dates = []
+    for p in points:
+        latitudes.append(p[0])
+        longitudes.append(p[1])
+        address = fake.address()
```

Comment:
`Faker` is used to create the addresses, but the addresses are never used since the `lat`, `lon` pair are generated programmatically. I recommend getting rid of the `Faker` dependency, and dropping the address generation code.